apps/prairielearn/src/lib/db-types.ts

```diff
@@ -580,7 +580,9 @@ export const CourseInstanceSchema = z.object({
   long_name: z.string().nullable(),
   self_enrollment_enabled: z.boolean(),
   self_enrollment_enabled_before_date: DateFromISOString.nullable(),
-  self_enrollment_requires_secret_link: z.boolean(),
+  // TODO: remove this column
+  self_enrollment_requires_secret_link: z.any(),
```

Comment:
Same question as in the migration guidelines, why not just drop it here?

Reply:
> Same question as in the migration guidelines, why not just drop it here?

The database schema needs to match up with the test in `db-types.test.ts`

Reply:
Ah, gross but reasonable, OK. There's no way for the tests to know about when somthing is in the middle of deletion.